File: ProjectTrackercs/PTWorkflow/ProjectWorkflow.vb
Public Class ProjectWorkflow
  Inherits SequentialWorkflowActivity

  Private _project As ProjectTracker.Library.Project

  Private Shared ReadOnly ProjectIdProperty As DependencyProperty = _
    DependencyProperty.Register("ProjectId", GetType(Guid), GetType(ProjectWorkflow), Nothing)

  Public Property ProjectId() As Guid
    Get
      Return CType(MyBase.GetValue(ProjectIdProperty), Guid)
    End Get
    Set(ByVal value As Guid)
      MyBase.SetValue(ProjectIdProperty, value)
    End Set
  End Property

  Private Sub login_ExecuteCode(ByVal sender As System.Object, ByVal e As System.EventArgs)

    ProjectTracker.Library.Security.PTPrincipal.Login("pm", "pm")

  End Sub

  Private Sub getProject_ExecuteCode(ByVal sender As System.Object, ByVal e As System.EventArgs)

    'Dim list As ProjectTracker.Library.ProjectList = ProjectTracker.Library.ProjectList.GetProjectList
    _project = ProjectTracker.Library.Project.GetProject(ProjectId) 'list(0).Id)

  End Sub


  Private Sub closeProject_ExecuteCode(ByVal sender As System.Object, ByVal e As System.EventArgs)

    _project.Ended = Today.ToString

  End Sub


  Private Sub saveProject_ExecuteCode(ByVal sender As System.Object, ByVal e As System.EventArgs)

    _project.Save()

  End Sub

End Class

File: ProjectTrackercs/PTWorkflow/ProjectWorkflow.cs
using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;
using ProjectTracker.Library;

namespace PTWorkflow
{
	public sealed partial class ProjectWorkflow: SequentialWorkflowActivity
	{
		public ProjectWorkflow()
		{
			InitializeComponent();
    }

    #region Dependency Properties

    private static DependencyProperty ProjectIdProperty = DependencyProperty.Register(
      "ProjectId", typeof(Guid), typeof(ProjectWorkflow), null);

    public Guid ProjectId
    {
      get { return (Guid)base.GetValue(ProjectIdProperty); }
      set { base.SetValue(ProjectIdProperty, value); }
    }

    //private static DependencyProperty ProjectProperty =
    //  DependencyProperty.Register("Project", typeof(Project), typeof(ProjectWorkflow), null);

    //public Project Project
    //{
    //  get { return (Project)base.GetValue(ProjectProperty); }
    //  set { base.SetValue(ProjectProperty, value); }
    //}
	
    #endregion

    #region Code Activities
    [System.Runtime.Serialization.OptionalField(VersionAdded=2)]
    private void getProject_ExecuteCode(object sender, EventArgs e)
    {
      //this.Project = Project.GetProject(ProjectId);
    }

    private void closeProject_ExecuteCode(object sender, EventArgs e)
    {
      //this.Project.Ended = DateTime.Today.ToString();
      //this.Project = this.Project.Save();
    }

    private void notifyResources_ExecuteCode(object sender, EventArgs e)
    {
      foreach (ProjectResource resource in this.getProject1.Project.Resources)
      {
        // notify each resource
      }
    }

    private void notifySponsor_ExecuteCode(object sender, EventArgs e)
    {
      // notify project sponsor
    }

    #endregion

  }

}
